demos/text_detection_demo/cpp/src/cnn.cpp

```diff
@@ -28,37 +28,35 @@ Cnn::Cnn(const std::string &model_path, Core & ie, const std::string & deviceNam
     if (inputInfo.size() != 1) {
         throw std::runtime_error("The network should have only one input");
     }
+    input_name_ = inputInfo.begin()->first;
     InputInfo::Ptr inputInfoFirst = inputInfo.begin()->second;
 
     SizeVector input_dims = inputInfoFirst->getInputData()->getTensorDesc().getDims();
+    if (input_dims.size() != 4) {
+        throw std::runtime_error("The network should have 4-dimensional input");
+    }
```

Comment:
I would not restrict the network to have 4 inputs. The easiest example is the same network but with LSTM decoder instead of GRU. LSTM would have 2 hidden tensors, and the whole network thus would have 5 inputs.

Reply:
This check is about one particular input dimensionality, not number of inputs. As for the number of inputs, this part corresponds to encoder, but not decoder

Reply:
Did not see this is the encoder. Agree.